net/src/client/tcp/actors/jsonrpc.rs

```diff
@@ -14,73 +15,110 @@ use async_jsonrpc_client::{
 };
 use futures::StreamExt;
 use futures_util::compat::Compat01As03;
+use rand::seq::SliceRandom;
 use serde::Serialize;
-use serde_json::{value, Value};
+use serde_json::value;
+
+pub use serde_json::Value;
 
 use super::Error;
 
+const DEFAULT_BACKOFF_TIME_MILLIS: u64 = 250;
+const MAX_BACKOFF_TIME_MILLIS: u64 = 15_000;
+
+struct Connection {
+    backoff: Duration,
+    socket: TcpSocket,
+    timestamp: Instant,
+    url: Arc<Mutex<String>>,
+}
+
 /// Json-RPC Client actor.
 ///
 /// Use this actor to send json-rpc requests over a websockets connection.
 pub struct JsonRpcClient {
     _handle: EventLoopHandle,
-    socket: TcpSocket,
     active_subscriptions: Arc<Mutex<HashMap<String, Subscribe>>>,
     pending_subscriptions: HashMap<String, Subscribe>,
-    url: String,
+    urls: Vec<String>,
     // Used to calculate the time since the last reconnection, and prevent multiple reconnections
     // in a short time interval
```

Comment:
This comment now belongs to `connection.timestamp`, right?

Reply:
Ooops I was a victim of refactoring without moving the comments. Nice catch!